### PR TITLE
Fix endpoint validation to allow returning string for all content types

### PR DESCRIPTION
**Pull request by benmccann**
Allows you to do:
```
export const get = () => {
    const body = '<foo />';
    return { status: 200, headers: { 'content-type': 'application/xml' }, body }
};
```

### CODE REVIEW

File: packages/kit/src/runtime/server/endpoint.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ function error(body) {
 	};
 }
```
Comment (JeanJPNM) [inline packages/kit/src/runtime/server/endpoint.js:12]: Adding a jsdoc here (`/** @param {string} s */`) removes the linting error

Comment (benmccann) [reply packages/kit/src/runtime/server/endpoint.js:12]: thanks

File: packages/kit/src/runtime/server/endpoint.js [hunk 2]
```diff
@@ -10,6 +10,11 @@ function error(body) {
...
 
+/** @param {string} s */
```
Comment (JeanJPNM) [inline packages/kit/src/runtime/server/endpoint.js:13]: Sorry, I messed up, `unknown` is a more appropriate type than `string` here.

Comment (benmccann) [reply packages/kit/src/runtime/server/endpoint.js]: guess I should have ran it locally :smile:

**Pull request merged**